src/netconfig.cpp

```diff
@@ -298,10 +298,11 @@ static void cmdVlan(Dbus& bus, Arguments& args)
         {
             puts("Can't delete a nonexistent interface.");
         }
-        else
+        else if (strstr(e.what(), "xyz.openbmc_project.Common.Error.InvalidArgument") != NULL)
         {
-            puts(e.what());
+            puts("Invalid VLAN Id provided! Allowed values according to 802.1Q-2005: 2-4094.");
```

Comment:
Wasn't sure how to properly word this message, decided to include mention of the limits.

Reply:
For the message, I'd propose this"
`Invalid VLAN ID. Must be [2 - 4094], see IEEE 802.1Q.`

However, I'm concerned about how you handle the error.
I believe the `InvalidArgument` exception may as well happen due to invalid interface specified, and then you'd still show this message that would be irrelevant.

I'd suggest printing just `Invalid argument` for this type of exception, and add an explicit check for the VLAN ID range before the `try`. Or may be even print `Invalid interface` here because it looks like that's the only way we could get here after an explicit check for VLAN ID.

Reply:
Added a helper function to check VLAN ID range and throw an exception with custom message.